modules/dbnd-airflow/src/dbnd_airflow/scheduler/single_dag_run_job.py

```diff
@@ -766,3 +801,45 @@ def heartbeat_callback(self, session=None):
             # self.executor.terminate()
             return
         self.terminating = True
+
+
+@provide_session
+def find_and_kill_zombies(args, session=None):
+    """
+    Find zombie DagRuns and TaskInstances that are in not end state and for which related
+    BackfillJob has failed.
+    """
+    print("Cleaning zombie tasks")
+    failed_jobs_id_qry = (
```

Comment:
we need to think about the query,  probably it's more about Running BaseJobs, or running DagRuns that have stollen BaseJobs

Reply:
Please elaborate more

Reply:
I mean not every BaseJob that failed is the source of zombie
I assume that if Job is marked as Failed -> it's has handled it's own dag runs.

We are more interested in Jobs that are Running with no heartbeat

Reply:
> I assume that if Job is marked as Failed -> it's has handled it's own dag runs.

No, this is a problem we are already seeing. I will add `or` clause to the query.